dask/dataframe/core.py

```diff
@@ -1510,7 +1510,9 @@ def to_bag(self, index=False):
         return to_bag(self, index)
 
     def _get_numeric_data(self, how='any', subset=None):
-
+        # If unknown dtype, need to infer from head.
+        if not self._known_dtype:
+            self.dtypes
```

Comment:
After sleeping on it, I think that it should be handled here. Most places we only need the column names, and not the dtypes when setting up the graph. This is one of the few places where the dtype is needed, so it makes sense to limit needed computation to here. Also, `_pd` as a property was just removed in #1017.